rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -931,17 +944,27 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
         if (picker == null) {
           return PickResult.withNoResult();
         }
-        return picker.pickSubchannel(args);
+        // Happy path goes through here
+        return pickResultOrError(picker.pickSubchannel(args), request.keyMap().get("server"));
```

Comment:
I don't think this is correct. This processes results from the child policies. We only want to adjust the status when we receive an error from the RLS server. If we did end up rewriting a status here, it would have misleading information. I suggest we do the status rewriting in the onError callback as the context is clear at that point, although the rewrite could be in the `response.hasError()` case below instead.

Rewriting the PickResult like here also has the difficulty that you have to check `isDrop()` and such as well. The only common case we re-create the PickResult is when it has a subchannel.

Reply:
The case that prompted this was the test lb_working_withDefaultTarget_noRlsResponse(RlsLoadBalancerTest.java:321) where all channels including the fallback are in failure state with a status of NOT_FOUND.  In that case the picker is a PickFirstLoadBalancer$Picker with subchannel=null and status NOT_FOUND.  response.hasData() == TRUE and response.hasError() == false.

![Screen Shot](https://user-images.githubusercontent.com/107004254/184420809-4987c40f-d27c-42a5-9eb5-a351f121c9a7.png)

Reply:
A subchannel shouldn't be failing with NOT_FOUND. Sounds like the test is misleading.